arch/sim/src/Makefile

```diff
@@ -141,7 +141,11 @@ endif
 
 ifeq ($(CONFIG_SIM_X11FB),y)
   HOSTSRCS += sim_x11framebuffer.c
+ifeq ($(CONFIG_HOST_MACOS),y)
+  STDLIBS += -L/opt/X11/lib -lX11 -lXext
+else
```

Comment:
let's remove #else block and share STDLIBS for both host.

Reply:
The directory is different on Mac and Linux. From macOS Catalina and above the X11 libraries will install to opt/X11/lib.
Only opt/X11/bin will be added to PATH so build system cannot find linker even aftermarket exporting /lib to PATH.

Reply:
yes, but we just need add the search path for macOS, but the library(e.g. -lX11) could be shared.

Reply:
If you have another ideas let me know, I had a hard time making the builds work on macOS and only this method looks the most efficient way ( works on Intel and Apple Silicon macs).

Reply:
@spi3ex basically what @xiaoxiang781216 is saying is:
#if BLABLABLA
     X += A + B
#else
    X += B
#endif

So, you can simplify it to:

X = B
#if BLABLABLA
    X += A
#endif

If the "-L/opt/X11/lib" and "-lX11 -lXext" order doesn't care, this suggestion should work fine.

Reply:
This Change is done. thanks.